src/drivers/main.cpp

```diff
@@ -1,59 +1,47 @@
+#include <cassert>
 #include <iostream>
+#include <precice/SolverInterface.hpp>
+#include <precice/Tooling.hpp>
+#include <stdexcept>
 #include <string>
-#include "precice/config/Configuration.hpp"
-#include "precice/impl/versions.hpp"
-#include "utils/assertion.hpp"
-#include "xml/Printer.hpp"
 
 void printUsage()
 {
-  std::cout << "Usage:\n\n";
-  std::cout << "Print XML reference      :  binprecice xml\n";
-  std::cout << "Print DTD for XML config :  binprecice dtd" << std::endl;
-  std::cout << "Print Markdown reference :  binprecice md" << std::endl;
+  std::cerr << "Usage:\n\n";
+  std::cerr << "Print XML reference      :  binprecice xml\n";
+  std::cerr << "Print DTD for XML config :  binprecice dtd\n";
+  std::cerr << "Print Markdown reference :  binprecice md\n";
+  std::cerr << "Print preCICE version    :  binprecice version\n";
```

Comment:
Why `binprecice version` and not `binprecice --version`? Or why not both? It is more complicated?

Reply:
`version` really is a command (_please give me the version info_) and not an option to an existing command (_please print the XML reference and additionally tell me the version info_).

I know that many tools implement the version as on option, however, many of these implementations are total non-sense. Docker for example prints the version and quits when passing `--version`, so it is a command disguised as an option. Also options normally come with a short form. For the version, this should be `-v`, which is also often used for short form of `--verbose`.

So, we could add the `--version` option, but it is a classical case of _high cost, low benefit_.

Reply:
But it should be an option to the application, right? An application does not need to have additional commands. I usually don't know anything about an application, but I know that I can type `application --version`.

Reply:
Well, not really. `application --version` means run the application (or default command) and additionally print the version. `application version` means, give me the version.
It is a weird confused convention (e.g [`git --version`](https://git-scm.com/docs/git#Documentation/git.txt---version) is implemented as an "alias" to `git version`, while `docker --version` means something different than `docker version`).

Anyhow, I'll add `--version` as an alias to move on.